Dockerfile

```diff
@@ -101,3 +101,23 @@ RUN cmake -DBUILD_CONTROLLERS="${BUILD_CONTROLLERS}" \
   && make -j all
 
 RUN CTEST_OUTPUT_ON_FAILURE=1 make test
+
+
+FROM development-dependencies as runtime-demonstrations
+
+WORKDIR /tmp/control_lib
+COPY ./source ./
+
+WORKDIR /tmp/control_lib/build
+RUN cmake -DBUILD_CONTROLLERS="ON" \
+    -DBUILD_DYNAMICAL_SYSTEMS="ON" \
+    -DBUILD_ROBOT_MODEL="ON" \
+    -DBUILD_TESTING="OFF" .. \
+  && make -j all \
+  && make install
+
+WORKDIR /tmp/
+COPY ./demos ./demos
+
+WORKDIR /tmp/demos/build
+RUN cmake .. && make -j all
```

Comment:
It might be nice to `rm` the control libraries source folder / build folder at the end to reduce image size slightly (since only the build targets are needed at the end). As a further step, you could move the executables out to a separate location and delete the demos build context too. But, that's a very minor improvement overall and it's fine how it is!

Reply:
I have found a way to do this simply by installing the `demos` package.